monai/networks/trt_compiler.py

```diff
@@ -0,0 +1,544 @@
+        """
+
+        profiles = []
+        if self.profiles:
+            for input_profile in self.profiles:
+                if isinstance(input_profile, Profile):
+                    profiles.append(input_profile)
+                else:
+                    p = Profile()
+                    for name, dims in input_profile.items():
+                        assert len(dims) == 3
+                        p.add(name, min=dims[0], opt=dims[1], max=dims[2])
+                    profiles.append(p)
+
+        build_args = self.build_args.copy()
+        build_args["tf32"] = self.precision != "fp32"
+        build_args["fp16"] = self.precision == "fp16"
+        build_args["bf16"] = self.precision == "bf16"
+
+        self.logger.info(f"Building TensorRT engine for {onnx_path}: {self.plan_path}")
+        network = network_from_onnx_path(onnx_path, flags=[trt.OnnxParserFlag.NATIVE_INSTANCENORM])
+        return engine_bytes_from_network(network, config=CreateConfig(profiles=profiles, **build_args))
+
+    def _build_and_save(self, model, input_example):
+        """
+        If TRT engine is not ready, exports model to ONNX,
+        builds TRT engine and saves serialized TRT engine to the disk.
+        Args:
+             input_example: passed to onnx.export()
+        """
+
+        if self.engine is not None:
+            return
+
+        export_args = self.export_args
+        dbs = self.dynamic_batchsize
+        if dbs:
+            if len(self.profiles) > 0:
+                raise ValueError("ERROR: Both dynamic_batchsize and input_profiles set for TrtCompiler!")
+            if len(dbs) != 3:
+                raise ValueError("dynamic_batchsize has to have len ==3 ")
+            profiles = {}
+            for id, val in input_example.items():
+                sh = val.shape[1:]
+                profiles[id] = [[dbs[0], *sh], [dbs[1], *sh], [dbs[2], *sh]]
+            self.profiles = [profiles]
+
+        if len(self.profiles) > 0:
+            export_args.update({"dynamic_axes": get_dynamic_axes(self.profiles)})
+
+        add_casts_around_norms(model)
+
+        if self.method == "torch_trt":
+            enabled_precisions = [torch.float32]
+            if self.precision == "fp16":
+                enabled_precisions.append(torch.float16)
+            elif self.precision == "bf16":
+                enabled_precisions.append(torch.bfloat16)
+            inputs = list(input_example.values())
+            ir_model = convert_to_torchscript(model, inputs=inputs, use_trace=True)
+            engine_bytes = torch_tensorrt.convert_method_to_trt_engine(
+                ir_model,
+                "forward",
+                inputs=inputs,
+                ir="torchscript",
+                enabled_precisions=enabled_precisions,
+                **export_args,
+            )
+        else:
+            if self.method == "onnx_dynamo":
+                dynamo = True
+                import torch_onnx
```

Comment:
should be optional_import here? And do we need include it in the requirement-dev list?

Reply:
Although it's experimental, we should also include it in the requirement-dev and add tests which can cover it.